transformer_engine/pytorch/ops/fuser.py

```diff
@@ -378,36 +353,33 @@ def __call__(
         *extra_inputs: torch.Tensor,
         basic_op_kwargs: Optional[list[dict[str, Any]]] = None,
     ) -> torch.Tensor | tuple[torch.Tensor, ...]:
+        # Verify extra input count
+        if len(extra_inputs) != self._num_extra_inputs:
+            raise ValueError(
+                f"Expected {self._num_extra_inputs} extra inputs but got {len(extra_inputs)}"
+            )
 
         # Initialization before forward pass
         for op in self._basic_ops:
             op.pre_forward()
 
         # Canonicalize op kwargs
         if basic_op_kwargs is None:
-            basic_op_kwargs = [{} for _ in range(len(self._basic_ops))]
-
-        # Flatten list of parameters
-        params = [param for op in self._basic_ops for param in op.parameters()]
+            basic_op_kwargs = [{}] * self._num_basic_ops
+        self._basic_op_kwargs = basic_op_kwargs
 
         # Fuser forward pass
-        is_grad_enabled = torch.is_grad_enabled()
-        if is_grad_enabled:
+        self._is_grad_enabled = torch.is_grad_enabled()
```

Comment:
[`torch.nn.Module` overrides `__setattr__`](https://github.com/pytorch/pytorch/blob/61e13782ddddf9e957c984ef11d7bb7643b871e7/torch/nn/modules/module.py#L1960), so this may have non-trivial overhead. I expect it would be better to pass `is_grad_enabled` and `basic_op_kwargs` directly into `_OperationFuserAutogradFunction`.